capstone-backend/src/main/java/com/google/sps/servlets/MeetingServlet.java

```diff
@@ -0,0 +1,142 @@
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(retrievedMeetings));
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    List<String> requiredFields = new ArrayList<String>(Arrays.asList("clubID", "startDateTime", "endDateTime"));
+    Meeting createdMeeting = (Meeting) Utility.postHelper(meetings, jsonObject, response, new GenericClass(Meeting.class),
+        requiredFields);
+    if (createdMeeting != null) {
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(createdMeeting));
+    }
+    TokenResponse token = new TokenResponse();
+    JsonObject tokenJson = jsonObject.get("token").getAsJsonObject();
+    token.setAccessToken(tokenJson.get("access_token").getAsString());
+    token.setTokenType(tokenJson.get("token_type").getAsString());
+    token.setScope(tokenJson.get("scope").getAsString());
+    token.setExpiresInSeconds(tokenJson.get("expires_in").getAsLong());
+    Credential credentials = createCredential(new NetHttpTransport(), new JacksonFactory(), token);
+    Calendar service = new Calendar.Builder(new NetHttpTransport(), new JacksonFactory(), credentials)
+        .setApplicationName("bookbook").build();
+    Set<String> keySet = jsonObject.keySet();
+    Event event = new Event();
+    if (keySet.contains("summary")) {
+      event.setSummary(jsonObject.get("summary").getAsString());
+    }
+    if (keySet.contains("location")) {
+      event.setLocation(jsonObject.get("location").getAsString());
+    }
+    if (keySet.contains("description")) {
+      event.setDescription(jsonObject.get("description").getAsString());
+    }
+    if (keySet.contains("startDateTime")) {
+      long startDateTime = jsonObject.get("startDateTime").getAsLong();
+      DateTime dateTime = new DateTime(startDateTime);
+      EventDateTime start = new EventDateTime().setDateTime(dateTime);
+      event.setStart(start);
+    }
+    if (keySet.contains("endDateTime")) {
+      long endDateTime = jsonObject.get("endDateTime").getAsLong();
+      DateTime dateTime = new DateTime(endDateTime);
+      EventDateTime end = new EventDateTime().setDateTime(dateTime);
+      event.setEnd(end);
+    }
+    if (keySet.contains("attendeeEmails")) {
+      JsonArray jsonArray = jsonObject.get("attendeeEmails").getAsJsonArray();
+      Type listType = new TypeToken<List<String>>() {}.getType();
+      List<String> attendeeEmails = gson.fromJson(jsonArray, listType);
+      for (String e : attendeeEmails) System.out.println(e);
+      event
+          .setAttendees(attendeeEmails.stream().map(e -> new EventAttendee().setEmail(e)).collect(Collectors.toList()));
+    }
+    EventReminder[] reminderOverrides = new EventReminder[] {
+        new EventReminder().setMethod("email").setMinutes(24 * 60),
```

Comment:
Really cool! Does this allow you to send an email reminder for the meeting 1 day in advance?

Reply:
It does, this will notify users in a fairly "typical" way about upcoming events.